luigi/contrib/redshift.py

```diff
@@ -52,7 +138,7 @@ class RedshiftTarget(postgres.PostgresTarget):
     use_db_timestamps = False
 
 
-class S3CopyToTable(rdbms.CopyToTable):
+class S3CopyToTable(rdbms.CopyToTable, _CredentialsMixin):
```

Comment:
Unfortunately I don't have the time to test it. I was doing something similar some time back and remember it failed because the ABC's from the base class where still abstract on instantiation. 
As this is just based on my memory your code might be correct already. Maybe someone else can help out?